test/e2e/cluster_upgrade.go

```diff
@@ -82,7 +82,7 @@ func ClusterUpgradeConformanceSpec(ctx context.Context, inputGetter func() Clust
 	It("Should create and upgrade a workload cluster and run kubetest", func() {
 		By("Creating a workload cluster")
 
-		var workerMachineCount int64 = 3
+		var workerMachineCount int64 = 2
```

Comment:
What about having 
```
var controlPlaneMachineCount int64 = 2
var machineDepoloymentMachineCount int64 = 2
var machinePoolMachineCount int64 = 2
```
So the math expression down in the code are more self explanatory

Reply:
Looks like:
```go
			ConfigCluster: clusterctl.ConfigClusterInput{
				LogFolder:                filepath.Join(input.ArtifactFolder, "clusters", input.BootstrapClusterProxy.GetName()),
				ClusterctlConfigPath:     input.ClusterctlConfigPath,
				KubeconfigPath:           input.BootstrapClusterProxy.GetKubeconfigPath(),
				InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
				Flavor:                   "upgrades",
				Namespace:                namespace.Name,
				ClusterName:              fmt.Sprintf("%s-%s", specName, util.RandomString(6)),
				KubernetesVersion:        input.E2EConfig.GetVariable(KubernetesVersionUpgradeFrom),
				ControlPlaneMachineCount: pointer.Int64Ptr(controlPlaneMachineCount),
				WorkerMachineCount:       pointer.Int64Ptr(machineDeploymentMachineCount),
			},
```

has only one WorkerMachineCount, so it would be a bit awkward to set it e.g. to machineDeploymentMachineCount, wdyt? (the constant for `controlPlaneMachineCount` makes sense in any case)

Reply:
oh I forgot that the template is using the same var in two places, which is kind of weird (might be worth a comment).
Similarly we could just add comment near each expression down in the code and that's will be enough to warn us when we go back to this sometimes in the future

Reply:
I think I found a good compromise.